Do not want to receive comments from action

Hello, I used this action, it is very convenient to replace now bot. I don't want to receive comments under commit every time I deploy to now. I removed github-token and it can be deployed to now, but ci reports an error. I hope that github-token can be turned into an optional item.

@amondnet same here, I removed GitHub token but every time I get ```##[error]Bad credentials
##[error]Node run failed with exit code 1``` though the app deploys well on Zeit.

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -14,4 +14,5 @@ jobs:
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           zeit-token: ${{ secrets.ZEIT_TOKEN }}
+          github-comment: false
           #ZEIT_TEAMID: amond

--- a/README.md
+++ b/README.md
@@ -24,15 +24,19 @@ This action make a ZEIT Now deployment with github actions.
 
 ### `zeit-team-id`
 
-This is required if your deployment is made on team project. example: `team_asdf1234`
+**optional** This is required if your deployment is made on team project and `github-comment` is true. example: `team_asdf1234`
+
+### `github-comment`
+
+**optional** This is required if you don't want to comment on pull request. default: true
 
 ### `github-token`
 
-**required** This is required to comment on pull request.
+**optional**  This is required if you want to comment on pull request.
 
 ### `now-args`
 
-This is optional args for `now` cli. Example: `--prod`
+**optional**  This is optional args for `now` cli. Example: `--prod`
 
 ## Outputs
 
@@ -72,10 +76,10 @@ jobs:
       - uses: actions/checkout@v1
       - uses: amondnet/now-deployment@v1
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          zeit-token: ${{ secrets.ZEIT_TOKEN }}
-          zeit-team-id: team_XXXXXXXXXXX
-          now-args: '--prod'
+          zeit-token: ${{ secrets.ZEIT_TOKEN }} # Required
+          github-token: ${{ secrets.GITHUB_TOKEN }} #Optional 
+          zeit-team-id: team_XXXXXXXXXXX #Optional 
+          now-args: '--prod' #Optional 
 ```
 
 

--- a/action.yml
+++ b/action.yml
@@ -14,10 +14,10 @@ inputs:
   github-comment:
     required: false
     default: 'true'
-    description: 'if you want to comment on pr or commit, set true, default: true'
+    description: 'if you want to comment on pr and commit, set true, default: true'
   github-token:
     required: false
-    description: 'if you want to comment on pr or commit, set token'
+    description: 'if you want to comment on pr and commit, set token'
   github-deployment:
     required: false
     default: 'false'

--- a/action.yml
+++ b/action.yml
@@ -11,9 +11,13 @@ inputs:
     description: ''
     required: false
     default: ''
-
+  github-comment:
+    required: false
+    default: 'true'
+    description: 'if you want to comment on pr or commit, set true, default: true'
   github-token:
-    required: true
+    required: false
+    description: 'if you want to comment on pr or commit, set token'
   github-deployment:
     required: false
     default: 'false'

--- a/index.js
+++ b/index.js
@@ -11,6 +11,7 @@ const zeitToken = core.getInput('zeit-token')
 const zeitTeamId = core.getInput('zeit-team-id')
 const nowArgs = core.getInput('now-args')
 const githubToken = core.getInput('github-token')
+const githubComment = core.getInput('github-comment')
 const githubDeployment = core.getInput('github-deployment')
 const workingDirectory = core.getInput('working-directory')
 
@@ -22,16 +23,21 @@ const zeitAPIClient = axios.create({
   },
 })
 
-const octokit = new github.GitHub(githubToken)
+let octokit
+if ( githubToken ) {
+  octokit = new github.GitHub(githubToken)
+}
 
 async function run () {
   await nowDeploy()
-  if (context.issue.number) {
-    core.info('this is related issue or pull_request ')
-    await createCommentOnPullRequest()
-  } else if (context.eventName === 'push') {
-    core.info('this is push event')
-    await createCommentOnCommit()
+  if ( githubComment && githubToken ) {
+    if (context.issue.number) {
+      core.info('this is related issue or pull_request ')
+      await createCommentOnPullRequest()
+    } else if (context.eventName === 'push') {
+      core.info('this is push event')
+      await createCommentOnCommit()
+    }
   }
 }
 
@@ -81,6 +87,9 @@ async function nowDeploy () {
 }
 
 async function listCommentsForCommit() {
+  if (!octokit) {
+    return;
+  }
   const {
     data: comments,
   } = await octokit.repos.listCommentsForCommit({
@@ -90,7 +99,9 @@ async function listCommentsForCommit() {
 }
 
 async function createCommentOnCommit () {
-
+  if (!octokit) {
+    return;
+  }
   const {
     data: comments,
   } = await octokit.repos.listCommentsForCommit({
@@ -170,7 +181,9 @@ async function createCommentOnCommit () {
 }
 
 async function createCommentOnPullRequest () {
-
+  if (!octokit) {
+    return;
+  }
   const {
     data: comments,
   } = await octokit.issues.listComments({

--- a/index.js
+++ b/index.js
@@ -38,6 +38,8 @@ async function run () {
       core.info('this is push event')
       await createCommentOnCommit()
     }
+  } else {
+    core.info('comment : disabled')
   }
 }
 

--- a/index.js
+++ b/index.js
@@ -11,8 +11,8 @@ const zeitToken = core.getInput('zeit-token')
 const zeitTeamId = core.getInput('zeit-team-id')
 const nowArgs = core.getInput('now-args')
 const githubToken = core.getInput('github-token')
-const githubComment = core.getInput('github-comment')
-const githubDeployment = core.getInput('github-deployment')
+const githubComment = core.getInput('github-comment') === 'true'
+const githubDeployment = core.getInput('github-deployment') === 'true'
 const workingDirectory = core.getInput('working-directory')
 
 const zeitAPIClient = axios.create({

--- a/package.json
+++ b/package.json
@@ -1,6 +1,16 @@
 {
   "name": "now-deployment",
   "private": true,
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/amondnet/now-deployment"
+  },
+  "author": {
+    "name": "Minsu Lee",
+    "email": "amond@amond.net",
+    "url": "https://amond.dev"
+  },
+  "version": "1.1.0",
   "main": "index.js",
   "scripts": {
     "start": "node ./index.js"

